contentcuration/contentcuration/frontend/channelList/views/KCard/index.vue

```diff
@@ -0,0 +1,215 @@
+          }"
+        />
+        <slot v-if="!thumbnailSrc" name="thumbnailPlaceholder"></slot>
+
+        <div>
+          <KRouterLink
+            :to="to"
+            :appearanceOverrides="{
+              'text-decoration': 'none',
+              'cursor': 'pointer',
+            }"
+            tabindex="-1"
+          >
+            <div
+              v-if="thumbnailDisplay === 'none'"
+              class="spacing"
+            >
+              <slot name="aboveTitle"></slot>
+              <slot name="title"></slot>
+              <slot name="belowTitle"></slot>
+            </div>
+            <KGrid
+              v-else
+              tabindex="-1"
+            >
+              <KGridItem
+                :layout12="{ span: isVerticalLayout ? 12 : 6 }"
+                :layout8="{ span: isVerticalLayout ? 8 : 4 }"
+                :layout4="{ span: 4 }"
+              >
+                <div class="spacing">
+                  <slot name="aboveTitle"></slot>
+                  <slot name="title"></slot>
+                  <slot name="belowTitle"></slot>
+                </div>
+              </KGridItem>
+
+              <KGridItem
+                v-if="layout === 'horizontal'"
+                :layout12="{ span: 6 }"
+                :layout8="{ span: 4 }"
+                :layout4="{ span: 4 }"
+              >
+                <KImg
+                  v-if="thumbnailDisplay === 'small'"
+                  :src="thumbnailSrc"
+                  :height="300"
+                  :width="300"
+                  :isDecorative="false"
+                  altText="thumbnail image"
+                  :appearanceOverrides="{
+                    objectFit: thumbnailScaleType,
+                    backgroundColor: $themeTokens.fineLine
+                  }"
+                  style="border-radius:20px;"
+                />
+                <slot v-if="!thumbnailSrc" name="thumbnailPlaceholder"></slot>
+
+              </KGridItem>
+            </KGrid>
+          </KRouterLink>
+
+
+          <div class="spacing">
+            <slot name="footer"></slot>
+          </div>
+        </div>
+      </div>
+    </div>
+  </BaseCard>
+
+</template>
+
+
+<script>
+
+  import KImg from '../KImg';
+  import BaseCard from './BaseCard.vue';
+
+  export default {
+    name: 'KCard',
+    components: { BaseCard, KImg },
+    props: {
+      title: {
+        type: String,
+        required: false,
+        default: null,
+      },
+      headingLevel: {
+        type: Number,
+        required: true,
+        validator(value) {
+          if (!value) {
+            console.error('Error: Prop headingLevel is required and can not be empty.');
+            return false;
+          }
+          return true;
+        },
+      },
+      titleLines: {
+        type: Number,
+        required: true,
+        default: 2,
+      },
+      to: {
+        type: Object,
+        required: true,
+        validator(value) {
```

Comment:
[Required] I don't think that custom validator for the basic presence of value is needed because when a prop is defined as `required`, Vue will take care by default to throw a warning in the console. You can remove.